src/docs/framework-integration/vue.md

```diff
@@ -8,50 +8,378 @@ contributors:
+
+```bash
+cd packages/
+npm init stencil components stencil-library
+cd stencil-library
+# Install dependencies
+npm install
+# of if you are using yarn
+yarn install
+cd ..
+lerna bootstrap
+```
+
+#### Creating a Vue Component Library
+
+> If you already have a Vue component library, skip this section.
+
+The first time you want to create the component wrappers, you will need to have a Vue library package to write to.
+
+Using Lerna and Vue's CLI, generate a workspace and a library for your Vue component wrappers:
+
+```bash
+lerna init
+npm install typescript @types/node --save-dev
+# or if you are using yarn
+yarn add typescript @types/node --dev
+
+lerna create vue-library
+# Follow the prompts and confirm
+cd packages/vue-library
+# Install Vue dependency
+npm install vue@3 --save-dev
+# or if you are using yarn
+yarn add vue@3 --dev
+# Add the stencil-library dependency
+lerna add stencil-library
+cd ../../
+lerna bootstrap
+```
+
+Lerna does not ship with a TypeScript configuration. At the root of the workspace, create a `tsconfig.json`:
 
-new Vue({
-  render: h => h(App)
-}).$mount('#app');
+```json
+{
+  "compilerOptions": {
+    "module": "commonjs",
+    "declaration": true,
+    "noImplicitAny": false,
+    "removeComments": true,
+    "noLib": false,
+    "emitDecoratorMetadata": true,
+    "experimentalDecorators": true,
+    "target": "es6",
+    "sourceMap": true,
+    "lib": ["es6"]
+  },
+  "exclude": ["node_modules", "**/*.spec.ts", "**/__tests__/**"]
+}
 ```
 
-The components should then be available in any of the Vue components
-```tsx
-render() {
-  return (
-    <div>
-      <test-stencil-component></test-stencil-component>
-    </div>
-  )
+Lerna does not create a `.gitignore` file, so we will manually create one:
+
+```ts
+node_modules/
+lerna-debug.log
+npm-debug.log
+packages/*/lib
+```
+
+In your `vue-library` project, create a project specific `tsconfig.json` that will extend the root config:
+
+```json
+{
+  "extends": "../../tsconfig.json",
+  "compilerOptions": {
+    "outDir": "./lib",
+    "lib": ["dom", "es2020"],
+    "module": "es2015",
+    "moduleResolution": "node",
+    "target": "es2017",
+    "skipLibCheck": true
+  },
+  "include": ["src"],
+  "exclude": ["node_modules"]
 }
 ```
 
-Vue provides several different ways to install and use the framework in an application. The above technique for integrating a Stencil custom element library has been tested on a Vue application that was created using the `vue-cli` with ES2015 and WebPack as primary options. A similar technique should work if the application was generated using other options.
+Update your `package.json`, adding the following options to the existing config:
+
+```json
+{
+  "types": "lib/index.d.ts",
+  "scripts": {
+    "test": "echo \"Error: run tests from root\" && exit 1",
+    "build": "npm run tsc",
+    "tsc": "tsc -p ."
+  },
+  "publishConfig": {
+    "access": "public"
+  }
+}
+```
```

Comment:
Consider using `diff` as the markdown type to help illustrate what's new:
```suggestion
```diff
{
+  "types": "lib/index.d.ts",
  "scripts": {
-   "test": "echo \"Error: run tests from root\" && exit 1"
+   "test": "echo \"Error: run tests from root\" && exit 1",
+    "build": "npm run tsc",
+    "tsc": "tsc -p ."
  },
+  "publishConfig": {
+    "access": "public"
+  }
}
```
```

Reply:
TIL about the `diff` markdown type, that's super cool!